components/AnimatedImage/library/src/desktopMain/kotlin/org/jetbrains/compose/animatedimage/LocalLoader.kt

```diff
@@ -0,0 +1,21 @@
+package org.jetbrains.compose.animatedimage
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.FileInputStream
+
+class LocalLoader(private val imageUrl: String) : AnimatedImageLoader {
+    var cachedBytes: ByteArray? = null
+
+    override suspend fun loadBytes(): ByteArray = withContext(Dispatchers.IO) {
+        var bytesArray: ByteArray? = cachedBytes
+
+        if (bytesArray == null) {
+            bytesArray = FileInputStream(imageUrl).readAllBytes() ?: byteArrayOf()
```

Comment:
nit: better to use readBytes extension, it returns `String` instead of `String!`